nautobot/utilities/renderers.py

```diff
@@ -0,0 +1,176 @@
+                restrict_form_fields(form, request.user)
+            elif view.action == "bulk_destroy":
+                if request.POST.get("_all"):
+                    if view.filterset_class is not None:
+                        pk_list = [obj.pk for obj in self.filterset_class(request.GET, model.objects.only("pk")).qs]
+                    else:
+                        pk_list = model.objects.values_list("pk", flat=True)
+                else:
+                    pk_list = request.POST.getlist("pk")
+                    initial = {
+                        "pk": pk_list,
+                        "return_url": view.get_return_url(request),
+                    }
+                    form = form_class(initial=initial)
+                table = view.table_class(view.queryset.filter(pk__in=pk_list), orderable=False)
+            elif view.action == "bulk_create":
+                form = view.get_form()
+            elif view.action == "bulk_update":
+                if request.POST.get("_all") and view.filterset_class is not None:
+                    pk_list = [obj.pk for obj in view.filterset_class(request.GET, view.queryset.only("pk")).qs]
+                else:
+                    pk_list = request.POST.getlist("pk")
+                if "_apply" in request.POST:
+                    form = form_class(model, request.POST)
+                    restrict_form_fields(form, request.user)
+                else:
+                    # Include the PK list as initial data for the form
+                    initial_data = {"pk": pk_list}
+                    # Check for other contextual data needed for the form. We avoid passing all of request.GET because the
+                    # filter values will conflict with the bulk edit form fields.
+                    # TODO: Find a better way to accomplish this
+                    if "device" in request.GET:
+                        initial_data["device"] = request.GET.get("device")
+                    elif "device_type" in request.GET:
+                        initial_data["device_type"] = request.GET.get("device_type")
+
+                    form = form_class(model, initial=initial_data)
+                    restrict_form_fields(form, request.user)
+                table = view.table_class(view.queryset.filter(pk__in=pk_list), orderable=False)
+        context = {
+            "obj": obj,
+            "object": instance,
+            "obj_type": view.queryset.model._meta.verbose_name,
+            "obj_type_plural": view.queryset.model._meta.verbose_name_plural,
+            "editing": obj.present_in_database,
+            "form": form,
+            "fields": form_class(model).fields if form_class else None,
+            "table": table if table else data.get("table", None),
+            "return_url": view.get_return_url(request, instance),
+            "verbose_name": view.queryset.model._meta.verbose_name,
+            "verbose_name_plural": view.queryset.model._meta.verbose_name_plural,
```

Comment:
Can we consolidate `obj_type[_plural]` and `verbose_name[_plural]` rather than having two sets of variables that mean the same exact thing?